From: Andrew Morton <akpm@linux-foundation.org>
Subject: panic-sys_info-factor-out-read-and-write-handlers-checkpatch-fixes
Date: Thu Oct 30 08:14:21 PM PDT 2025

WARNING: struct ctl_table should normally be const
#24: FILE: lib/sys_info.c:46:
+static int sys_info_write_handler(struct ctl_table *table,

WARNING: struct ctl_table should normally be const
#43: FILE: lib/sys_info.c:65:
+static int sys_info_read_handler(struct ctl_table *table,

total: 0 errors, 2 warnings, 99 lines checked

NOTE: For some of the reported defects, checkpatch may be able to
      mechanically convert to the typical style using --fix or --fix-inplace.

./patches/panic-sys_info-factor-out-read-and-write-handlers.patch has style problems, please review.

NOTE: If any of the errors are false positives, please report
      them to the maintainer, see CHECKPATCH in MAINTAINERS.

Please run checkpatch prior to sending patches

Cc: Andy Shevchenko <andriy.shevchenko@linux.intel.com>
Cc: Feng Tang <feng.tang@linux.alibaba.com>
Cc: Petr Mladek <pmladek@suse.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 lib/sys_info.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- a/lib/sys_info.c~panic-sys_info-factor-out-read-and-write-handlers-checkpatch-fixes
+++ a/lib/sys_info.c
@@ -43,7 +43,7 @@ unsigned long sys_info_parse_param(char
 
 #ifdef CONFIG_SYSCTL
 
-static int sys_info_write_handler(struct ctl_table *table,
+static int sys_info_write_handler(const struct ctl_table *table,
 				  void *buffer, size_t *lenp, loff_t *ppos,
 				  unsigned long *si_bits_global)
 {
@@ -62,7 +62,7 @@ static int sys_info_write_handler(struct
 	return 0;
 }
 
-static int sys_info_read_handler(struct ctl_table *table,
+static int sys_info_read_handler(const struct ctl_table *table,
 				 void *buffer, size_t *lenp, loff_t *ppos,
 				 unsigned long *si_bits_global)
 {
_
